tests/unit/parallel/algorithms/copyif_random.cpp

```diff
@@ -21,7 +23,7 @@
 ////////////////////////////////////////////////////////////////////////////
 unsigned int seed = std::random_device{}();
 std::mt19937 gen(seed);
-std::uniform_int_distribution<> dis(0,std::numeric_limits<int>::max());
+std::uniform_int_distribution<> dis(0,boost::integer_traits<int>::max());
```

Comment:
Why did you go away from `std::numeric_limits`?

Reply:
Because Circle CI was failing `violation of Boost min/max guidelines`

Reply:
Yah, sure. If you look around you'll find this: http://www.boost.org/doc/libs/1_34_0/more/lib_guide.htm (amongst other things). Essentially the Boost guidelines say that in order to avoid `min` or `max` being expanded as a macro you need to accordingly protect all of their uses. In  case of `std::numeric_limits<T>::max()` you would have to write `(std::numeric_limits<T>::max)()`.

Reply:
Thanks for correcting me :)
I will fix it right away.